### PR TITLE
Migrate /about/leadership/ page to Wagtail (Fixes #14801)

### LINKED ISSUE #14801: Migrate /about/leadership/ page to Wagtail
## Description

This is a good-first-surface to power using Wagtail as it's `en-US` only (no L10n requirement) and is updated regularly.

CMS requirements are relatively straight forward:

- Add / Edit / Remove leadership / board members based on groupings.
- Upload profile images.
- Add guard rails around required / optional fields, number of items etc.

We can add things like publishing schedules at a later step.

---

#### Success Criteria
- [ ] Current page / design is fully editable in Wagtail.

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

Migrates `/about/leadership/` page to Wagtail

## Significant changes and points to review

This does not yet remove the old leadership page / template. That will be done after we have rolled this out and have data in the prod DB.

## Issue / Bugzilla link

#14801

## Testing

> [!NOTE]
> A good test is to use some data from https://www.mozilla.org/en-US/about/leadership/ to create some different sections / groups.

1. Run `make preflight`.
2. Run `npm start`.
3. Auth via http://localhost:8000/cms-admin/.
4. Create an `StructuralPage` with a slug of `about`.
5. Create a child `LeadershipPage` under the `StructuralPage` you created in the prior step, with a slug of `leadership`.
6. Enter some data and preview the page.
7. Publish the page.
8. Go to http://localhost:8000/en-US/about/leadership/, and you should now see the wagtail page instead of the hard-coded page.

### CODE REVIEW

File: bedrock/mozorg/models.py
Comment (alexgibson) [inline bedrock/mozorg/models.py:73]: **issue:** I'm not sure why this doesn't work? Might it be an artifact from `AbstractBedrockCMSPage`?

**Comment by alexgibson** [comment]
@stevejalim I'm going to continue building out some content to test locally, but I think this is at a good enough spot for a first pass.

**Comment by alexgibson** [comment]
@stevejalim added a commit to use the new `prefers_cms` decorator. Seems to work! 💯

**Comment by alexgibson** [comment]
OK, I think this is ready for review

**Review by stevejalim — COMMENTED** [review]
Separating out the code review from the test-driving review.

This is lovely, lovely work. The Wagtail-specific stuff is clear and well executed - and a real pleasure to review. Most of my comments are questions or notes

**Comment by stevejalim** [comment]
> wanted you to be aware in case we consider it a blocker (I do)

To clarify - it's definitely an issue to sort, but it's not a blocker to merging this PR, just a blocker to revealing/using this page in production

File: bedrock/cms/models/pages.py
Comment (alexgibson) [reply bedrock/cms/models/pages.py:45]: I think you suggested this for a reason (something to do with migrations?)

File: bedrock/mozorg/blocks/leadership.py
Comment (alexgibson) [reply bedrock/mozorg/blocks/leadership.py:24]: Yeah, i guess we could always change this later, but for now I stuck with the assets.m.o link mainly because these zip files are quite large?

File: bedrock/mozorg/templates/mozorg/about/leadership/blocks/leadership_block.html
Comment (alexgibson) [reply bedrock/mozorg/templates/mozorg/about/leadership/blocks/leadership_block.html:7]: The ID is required for the modal JS, and we only want to to trigger if the person has an actual bio.

**Comment by alexgibson** [comment]
> I have wagtail-localize active and the lang selector in the bottom right shows all the langs that are available in the system, even though this page only exists in the en-US tree. If I pick, say, Polish, I get the en-US content under a /pl/ path. This looks like a separate issue and nothign at all to do with the changes here, but wanted you to be aware in case we consider it a blocker (I do)

Yeah, we should fix this but agree it can be done separately.

**Comment by stevejalim** [comment]
@alexgibson The extra line below the header was down to me failing to recompile the CSS properly first, sorry!

File: tests/playwright/specs/leadership.spec.js
Comment (alexgibson) [reply tests/playwright/specs/leadership.spec.js:24]: I've updated the test here so it should pass for both the hard-coded page and the CMS page (it assumes at least our CEO will have data entered). So, things should pass OK after merging still because we're using `prefers_cms`, right?

File: bedrock/cms/models/pages.py
Comment (stevejalim) [reply bedrock/cms/models/pages.py:45]: Looks like it dates from when you wanted to make a `AboutPage(StructuralPage)` class and it was creating an extra layer of multi-table inheritance - convo is here https://mozilla.slack.com/archives/D02GDP6PAJG/p1720613090788999

Based on this changeset tho, and the instructions, i don't see you using a custom AboutPage after all, so we could back out the changes here -- or leave them in case we do find we want to have other kinds of StructuralPage subclasses

**Comment by alexgibson** [comment]
@stevejalim thanks for the review! I've added another commit which I think should address the comments so far.

File: bedrock/cms/models/pages.py
Comment (alexgibson) [reply bedrock/cms/models/pages.py:45]: Yeah I wasn't sure what was really needed / expected here, happy to revert if we think defining `AboutPage(StructuralPage)` is overkill.

File: tests/playwright/specs/leadership.spec.js
Comment (stevejalim) [reply tests/playwright/specs/leadership.spec.js:24]: Yeah, once the page is live in the environment where playwright runs, I can see that it should pass, yeah.

**Review by stevejalim — APPROVED** [review]
🎉 Fabulous stuff - nice one! r+ 🎉

**Comment by alexgibson** [comment]
@stevejalim thanks for the review! Whilst this should be safe to merge, I'm gonna hold off until Tuesday next week, after the upcoming EU campaign page has launched - just to be safe 🤞

**Pull request merged**
